src/dialect/mod.rs

```diff
@@ -71,6 +71,10 @@ pub trait Dialect: Debug + Any {
     fn supports_filter_during_aggregation(&self) -> bool {
         false
     }
+    /// Does the dialect supports ARRAY_AGG() [WITHIN GROUP (ORDER BY)] or ARRAY_AGG([ORDER BY])
+    fn supports_within_after_array_aggregation(&self) -> bool {
```

Comment:
The comment is confusing.

The comment make it sounds like if any of both is valid, it will return true.

Could be something like

```rust
/// Returns true if the dialect supports ARRAY_AGG() [WITHIN GROUP (ORDER BY)] expressions.
///
/// Otherwise, the dialect should expect an `ORDER BY` without the `WITHIN GROUP` clause, e.g. `ANSI` [(1)].
/// [(1)]: https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#array-aggregate-function
```

Having the function here is not my usual approach, as it will require maintenance over other dialects individually, but I don't think it's wrong.

Reply:
Also, @SuperBo, one thing:
- Generic dialect should be the most permissive one so, if you actually can have both structures in a query, generic dialect should allow both. But considering my previous comment, I'd allow the most common one, maybe?

@alamb what do you think? This is a divergence from dialects. I'd go for the one with more dialects, so the generic is as generic as possible.

Reply:
I think you are right, the ANSI one should be default option.

Reply:
@AugustoFKL, are we good to go with this or should we change to more generics behavior?